src/main/java/de/tum/in/www1/artemis/repository/FileUploadExerciseRepository.java

```diff
@@ -34,6 +35,20 @@ public interface FileUploadExerciseRepository extends JpaRepository<FileUploadEx
     @EntityGraph(type = LOAD, attributePaths = { "teamAssignmentConfig", "categories", "learningGoals" })
     Optional<FileUploadExercise> findWithEagerTeamAssignmentConfigAndCategoriesAndLearningGoalsById(Long exerciseId);
 
+    @Query("""
+            SELECT e
+            FROM Course c
+            LEFT JOIN  c.exercises e
+            LEFT JOIN FETCH e.studentParticipations p
+            LEFT JOIN FETCH p.submissions s
+            LEFT JOIN FETCH s.results r
+            Where c.id = :courseId
+            AND p.student.id = :userId
+            AND TYPE(e) = FileUploadExercise
+            """)
+    Set<FileUploadExercise> getAllFileUploadExercisesWithEagerParticipationsSubmissionsAndResultsOfUserFromCourseByCourseAndUserId(@Param("courseId") long courseId,
+            @Param("userId") long userId);
```

Comment:
Can we move this method into the ExerciseRepository and provide an extra argument of the exercise type?

Reply:
Or fetch all exercises in createDataExport and create a map from the exercise type to a list of each exercise there. This might even be more performant...

Reply:
I'll try the latter approach.